SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -432,28 +450,22 @@ void CSolver::InitiatePeriodicComms(CGeometry *geometry,
 
         /*--- Store angles separately for clarity. Compute sines/cosines. ---*/
 
-        Theta    = angles[0];      Phi = angles[1];     Psi = angles[2];
-        cosTheta = cos(Theta);  cosPhi = cos(Phi);   cosPsi = cos(Psi);
-        sinTheta = sin(Theta);  sinPhi = sin(Phi);   sinPsi = sin(Psi);
+        su2double Theta = angles[0];
+        su2double Phi = angles[1];
+        su2double Psi = angles[2];
 
         /*--- Compute the rotation matrix. Note that the implicit
          ordering is rotation about the x-axis, y-axis, then z-axis. ---*/
 
-        rotMatrix[0][0] = cosPhi*cosPsi;
-        rotMatrix[1][0] = cosPhi*sinPsi;
-        rotMatrix[2][0] = -sinPhi;
-
-        rotMatrix[0][1] = sinTheta*sinPhi*cosPsi - cosTheta*sinPsi;
-        rotMatrix[1][1] = sinTheta*sinPhi*sinPsi + cosTheta*cosPsi;
-        rotMatrix[2][1] = sinTheta*cosPhi;
-
-        rotMatrix[0][2] = cosTheta*sinPhi*cosPsi + sinTheta*sinPsi;
-        rotMatrix[1][2] = cosTheta*sinPhi*sinPsi - sinTheta*cosPsi;
-        rotMatrix[2][2] = cosTheta*cosPhi;
+        if (nDim==2) {
```

Comment:
can the nDim checks be put inside the function efficiently? Feels like they could be collapsed with an extra arg or template param

Reply:
The implementation of Rotate determines a template nDim (for unrolling) from the size of the rotation matrix that's why there is a 2D and 3D copy... In short I don't see an easy 0 cost alternative.